util/random/rand.go

```diff
@@ -71,8 +76,60 @@ func (r *randstream) XORKeyStream(dst, src []byte) {
 	}
 }
 
-// New returns a new cipher.Stream that gets random data from Go's crypto/rand package.
+// READER_BYTES is how many bytes we expect from each source
+const READER_BYTES = 32
```

Comment:
I don't see what this adds to the public interface of this package, so please just move it close to where it is used, and not in package global scope.

Reply:
It adds information about how much data will be read from the different readers. The user of this API needs to know how many bytes should its reader have. 
Just to be clear, I explicitely asked @PizzaWhisperer to put it as a variable there so blame it on me ;) 
No worries to change it to a local variable but then we need to explicitely mention that same number in the comment, otherwise the API is un-specified.